// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: category/category.proto

package category

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// Validate checks the field values on FindRequest with the rules defined in
// the proto definition for this message. If any rules are violated, an error
// is returned.
func (m *FindRequest) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for Html

	return nil
}

// FindRequestValidationError is the validation error returned by
// FindRequest.Validate if the designated constraints aren't met.
type FindRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e FindRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e FindRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e FindRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e FindRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e FindRequestValidationError) ErrorName() string { return "FindRequestValidationError" }

// Error satisfies the builtin error interface
func (e FindRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sFindRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = FindRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = FindRequestValidationError{}

// Validate checks the field values on FindResponse with the rules defined in
// the proto definition for this message. If any rules are violated, an error
// is returned.
func (m *FindResponse) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for CategoryId

	return nil
}

// FindResponseValidationError is the validation error returned by
// FindResponse.Validate if the designated constraints aren't met.
type FindResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e FindResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e FindResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e FindResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e FindResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e FindResponseValidationError) ErrorName() string { return "FindResponseValidationError" }

// Error satisfies the builtin error interface
func (e FindResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sFindResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = FindResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = FindResponseValidationError{}

// Validate checks the field values on CategoryItem with the rules defined in
// the proto definition for this message. If any rules are violated, an error
// is returned.
func (m *CategoryItem) Validate() error {
	if m == nil {
		return nil
	}

	// no validation rules for Id

	// no validation rules for Title

	// no validation rules for Slug

	return nil
}

// CategoryItemValidationError is the validation error returned by
// CategoryItem.Validate if the designated constraints aren't met.
type CategoryItemValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e CategoryItemValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e CategoryItemValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e CategoryItemValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e CategoryItemValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e CategoryItemValidationError) ErrorName() string { return "CategoryItemValidationError" }

// Error satisfies the builtin error interface
func (e CategoryItemValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sCategoryItem.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = CategoryItemValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = CategoryItemValidationError{}
